### DISCUSSION TITLE
Is Privately Publish possible?

### DISCUSSION THREAD

**Discussion by kayes99** [community/community#123365] [New to GitHub]
Is there any way to publish my website from GitHub so no one can copy or download my code?

**Comment by murtuja43** [comment]
Unfortunately, if you publish your website code on GitHub and make the website publicly accessible, anyone can view and potentially copy your code. Here's why:

Open Source Nature: GitHub is built on the concept of open collaboration. Public repositories allow others to see, learn from, and potentially contribute to the code.
However, there are ways to make it more difficult to directly copy your code and some strategies to protect your intellectual property:

Obfuscation (Limited Effectiveness):

You can obfuscate your code. This scrambles the code while maintaining functionality, making it harder to read and understand. However, with enough effort, a determined individual can de-obfuscate the code.
Focus on Website Functionality:

If your website's core value lies in its functionality and user experience, consider focusing on making it difficult to replicate that experience by another website.
Server-Side Logic:

Move sensitive logic or functionalities to a server-side language like Python or Node.js. This keeps that code private while the website's user interface remains accessible.
Focus on Content:

If your website is content-driven, like a blog or portfolio, focus on creating unique and valuable content that would be difficult to replicate elsewhere.
Licensing:

Consider using a license like MIT or Apache. This allows others to use your code under certain conditions, potentially giving you some control over its use.
Important Note:  Licenses aren't foolproof and don't prevent copying. They simply establish terms for how others can use your code.

Alternative: Code Secrecy

If complete code secrecy is crucial, consider hosting your website on a private platform where you control access. This eliminates the public visibility aspect of GitHub.

**Comment by krishealty** [comment]
Yes you can but unfortunately you need to buy the pro for it.

GitHub Pages is available in public repositories with GitHub Free and GitHub Free for organizations, and in public and private repositories with GitHub Pro, GitHub Team, GitHub Enterprise Cloud, and GitHub Enterprise Server. For more information, see [Github Plans.](https://github.com/pricing)

Regards.

**Comment by CiceroLino** [comment]
Hi, interesting question.

So, more or less. You can put your code on github and use other service to publish it. But if your web site are made only html and css. We still can see your code
